dist/PreserveModuleNamePlugin.js

```diff
@@ -73,6 +73,29 @@ function getPreservedModules(modules) {
                 m[exports.preserveModuleName] = req;
                 return true;
             }
+            // Since its relative, link it with the module alias if possible
+            const reasonRequest = r.module && r.module.rawRequest;
+            const moduleDependency = r.dependency && r.dependency.module;
+            const moduleRequest = moduleDependency && removeLoaders(moduleDependency.request);
+            if (req && aliases && reasonRequest && moduleRequest) {
+                const aliasMatches = Object.keys(aliases).filter((alias) => {
+                    const aliasPath = aliases[alias];
+                    const matchPath = !!(moduleRequest.match(new RegExp(`^${aliasPath}`)));
+                    const matchKey = !!(reasonRequest.match(new RegExp(`^${alias}`)));
```

Comment:
If you're not interested in the results, just whether the regex matches, using `.search()` is more self explanatory.
All RE functions like `match`, `exec` or `search` can take a string, no need to `new RegExp` in there.